nixos/modules/services/x11/desktop-managers/elementary.nix

```diff
@@ -0,0 +1,201 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.xserver.desktopManager.elementary;
+
+  nixos-gsettings-desktop-schemas = pkgs.elementary.elementary-gsettings-schemas.override {
+    extraGSettingsOverridePackages = cfg.extraGSettingsOverridePackages;
+    extraGSettingsOverrides = cfg.extraGSettingsOverrides;
+  };
+
+in
+
+{
+  options = {
+
+    services.xserver.desktopManager.elementary = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the elementary desktop manager";
+      };
+
+      sessionPath = mkOption {
+        default = [];
+        example = literalExample "[ pkgs.gnome3.gpaste ]";
+        description = ''
+          Additional list of packages to be added to the session search path.
+          Useful for GSettings-conditional autostart.
+
+          Note that this should be a last resort; patching the package is preferred (see GPaste).
+        '';
+        apply = list: list ++
+        [
+          pkgs.elementary.elementary-dpms-helper
+          pkgs.elementary.elementary-settings-daemon # TODO: I shouldn't need to do this
+          pkgs.elementary.pantheon-agent-geoclue2
+          pkgs.gnome3.evolution-data-server
+        ];
+      };
+
+      extraGSettingsOverrides = mkOption {
+        default = "";
+        type = types.lines;
+        description = "Additional gsettings overrides.";
+      };
+
+      extraGSettingsOverridePackages = mkOption {
+        default = [];
+        type = types.listOf types.path;
+        description = "List of packages for which gsettings are overridden.";
+      };
+
+      debug = mkEnableOption "gnome-session debug messages";
+
+    };
+
+    environment.elementary.excludePackages = mkOption {
+      default = [];
+      example = literalExample "[ pkgs.elementary.elementary-camera ]";
+      type = types.listOf types.package;
+      description = "Which packages elementary should exclude from the default environment";
+    };
+
+  };
+
+
+  config = mkIf cfg.enable {
+
+    services.xserver.displayManager.extraSessionFilePackages = [ pkgs.elementary.elementary-session-settings ];
+
+    # Ensure lightdm is used when Pantheon is enabled
+    # Without it screen locking will be nonfunctional because of the use of lightlocker
+    services.xserver.displayManager.lightdm.enable = mkDefault true;
+    services.xserver.displayManager.lightdm.greeters.elementary.enable = mkDefault true;
+
+    # If not set manually Pantheon session cannot be started
+    # Known issue of https://github.com/NixOS/nixpkgs/pull/43992
+    services.xserver.desktopManager.default = mkForce "pantheon";
+
+    services.xserver.displayManager.sessionCommands = ''
+      if test "$XDG_CURRENT_DESKTOP" = "Pantheon"; then
```

Comment:
@jtojnar  I recall you mentioning that I needed to change this and now I also see https://github.com/NixOS/nixpkgs/commit/cdd266c73bee26b6cd7e092da9a4cc0f34237434

Reply:
The variable can contain multiple desktop names, I think I started the upstream sessions because of Flashback, which has something like `Flashback:GNOME`, not sure what Pantheon does.

I am not particularly sure why did we remove the check altogether instead of using https://github.com/jtojnar/nixpkgs/commit/f08717f98dfd19ab848cb29d7c507cbd9a27bcd7. Perhaps so it could be re-used from Pantheon?

https://github.com/NixOS/nixpkgs/issues/39871#issuecomment-410655127
https://github.com/NixOS/nixpkgs/pull/44497#issuecomment-410683470